### PR TITLE
Reflect donations endpoint changes

### PR DESCRIPTION
**Pull request by ap-justin**
Closes BG-1249

NOTE: changes for use of `finalAmountUsd` would be on another PR to easily distinguish the changes made to reflect the new donations endpoint

## Explanation of the solution
* reflect changes in 
  * #546
* devops: authorizers are now applied to `GET (v1 | staging) / donations` 
* kindly see change comments for details

## Instructions on making this work

- run `yarn` or `yarn install` to install npm dependencies
- run `yarn run test --watchAll` to verify all tests still pass
- (optional) run `yarn run build` to verify the build passes
- run `yarn start` to start the webapp
-

## UI changes for review

When major UI changes are introduced with a PR, please include links to URLS to compare or screenshots demonstrating the difference and notify on design changes

### CODE REVIEW

File: src/services/aws/aws.ts [hunk 1]
```diff
@@ -199,6 +202,18 @@ export const aws = createApi({
         };
       },
     }),
+    donations: builder.query<
```
Comment (ap-justin) [inline src/services/aws/aws.ts:205]: endpoint definitions has been moved from `services/apes` to `services/ap` as lambda also exist in AP

File: src/types/aws/ap/donations.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
+import { ChainID } from "types/chain";
```
Comment (ap-justin) [inline src/types/aws/ap/donations.ts:1]: types are also moved from `apes` to `aws`

File: src/services/aws/usePaginatedDonations.ts [hunk 3]
```diff
@@ -107,6 +93,10 @@ export default function usePaginatedDonationRecords<T extends Args>(args: T) {
     query,
     loadNextPage,
     onQueryChange: setQuery,
+    status: originalArgs?.status || "final",
+    setStatus: (status: DonationsQueryParams["status"]) =>
+      //reset page when switching status (changes table source)
+      setParams((prev) => ({ ...prev, status, page: 1 })),
```
Comment (misicnenad) [inline src/services/aws/usePaginatedDonations.ts:99]: Don't know what the desired behavior is here, but these changes no longer preserve previously loaded table items (loaded with "load more" button) for Donations table, which is the current behavior on RC-v1.9.2/staging.
Repro:
1. Sign in with account that has > 10 received donations
2. Go to Donations page
3. Wait for received donations to load
4. Click on "load more"
5. wait for additional donations to load
6. switch to "Pending" donations
7. wait for donations to load
8. switch back to "Received"
9. **expected:** previously loaded donations are preserved <- this is behavior on https://staging.better.giving/donations
**actual:** table is reset, additional donations need to be reloaded with "load more" btn

File: src/pages/Donations/Table.tsx [hunk 4]
```diff
@@ -100,35 +100,39 @@ export default function Table({
               <Link
                 to={`${
                   appRoutes[
-                    row.chainId === chainIds.juno ? "profile" : "marketplace"
+                    row.viaId === chainIds.juno ? "profile" : "marketplace"
                   ]
                 }/${row.id}`}
```
Comment (misicnenad) [inline src/pages/Donations/Table.tsx:105]: This now results in corrupt URLs:
image

Should this be `row.recipientId`?

**Review by misicnenad — CHANGES_REQUESTED** [review]

File: src/pages/Admin/Charity/Donations/DonationsTable.tsx [hunk 5]
```diff
@@ -79,19 +89,31 @@ export default function DonationsTable({ classes = "" }) {
...
   { key: "symbol", label: "Currency" },
   { key: "date", label: "Date" },
-  { key: "hash", label: "Transaction Hash" },
+  { key: "id", label: "Transaction Hash" },
 ];
 
+/** fill undefined with "" */
+export function fill<T extends object>(
+  obj: T
+): { [K in keyof T]-?: NonNullable<T[K]> } {
```
[the remark this answers is not in the document]
Comment (ap-justin) [reply src/pages/Admin/Charity/Donations/DonationsTable.tsx:104]: `NonNullable` only removes `undefined | null` in a union e.g. `x?: undefined | null | value ---> x?:value`,
`[]-?` operator first converts `x?:value --> x: undefined | value`

File: src/services/aws/usePaginatedDonations.ts (hunk 3 above)
Comment (ap-justin) [reply src/services/aws/usePaginatedDonations.ts:99]: I think the previous one is desirable (avoids loading again). Resetting page I guess is unncessarily worrying the transition from `loaded:page-3` to `pending:page-3`

Comment (ap-justin) [reply src/services/aws/usePaginatedDonations.ts]: fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2853#:~:text=remove%20page%20reset-,00946c4,-use%20recipient%20id

File: src/pages/Donations/Table.tsx (hunk 4 above)
Comment (ap-justin) [reply src/pages/Donations/Table.tsx]: fixed https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2853/commits/826ff9942672caa2d88a5bdea265b3f81ca7e964

File: src/pages/Admin/Charity/Donations/DonationsTable.tsx [hunk 6]
```diff
@@ -79,19 +89,31 @@ export default function DonationsTable({ classes = "" }) {
...
-  { key: "amount", label: "Amount" },
+  { key: "initAmount", label: "Amount" },
   { key: "symbol", label: "Currency" },
   { key: "date", label: "Date" },
-  { key: "hash", label: "Transaction Hash" },
+  { key: "id", label: "Transaction Hash" },
 ];
 
+/** fill undefined with "" */
+export function fill<T extends object>(
```
[the remark this answers is not in the document]
Comment (ap-justin) [reply src/pages/Admin/Charity/Donations/DonationsTable.tsx]: fixed: https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/2853/commits/e3c32dc1c5623430d85203a8de988a3b4a63ae0e

**Review by SovereignAndrey — APPROVED** [review]

File: src/pages/Admin/Charity/Donations/DonationsTable.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (misicnenad) [reply src/pages/Admin/Charity/Donations/DonationsTable.tsx:104]: Oh I see, makes sense.

File: src/services/aws/usePaginatedDonations.ts (hunk 3 above)
Comment (misicnenad) [reply src/services/aws/usePaginatedDonations.ts]: The behavior is still altered, the table gets reset after switching tabs.

**Pull request merged**